src/index.ts

```diff
@@ -28,6 +28,44 @@ export const getCustomErrorMessage = (
   return possibleProgramErrors[errorNumber] || null;
 };
 
+const encodeURL = (baseUrl: string, searchParams: Record<string, string>) => {
+  // This was a little new to me, but it's the
+  // recommended way to build URLs with query params
+  // (and also means you don't have to do any encoding)
+  // https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams
+  const url = new URL(baseUrl);
+  url.search = new URLSearchParams(searchParams).toString();
+  return url.toString();
+};
+
+export const getExplorerLink = (
+  linkType: "transaction" | "tx" | "address" | "block",
+  id: string,
+  cluster: Cluster | "localnet" = "mainnet-beta",
```

Comment:
Bit of an edge case, but should we add support for other custom cluster urls?
Someone might have a specific url or part they are want to access a cluster

Reply:
Interesting thought. Specifying values gets us validation in more cases, but yes if someone is using a custom cluster URL TS compilation will fail. Does that change anything for you? I could go either way so LMK what you prefer.

Reply:
Yeah, that's fair. its an edge case like I mentioned.
Let's not worry about it for now. Worst case, we come back to it later if people start requesting it :)